File: ts/src/search/chunker.ts
/**
 * Split text into overlapping chunks for embedding.
 *
 * Uses a simple token-count approximation (split on whitespace).
 * For production use with precise token counting, wrap this with a
 * proper tokenizer (tiktoken, etc.).
 */
export function splitIntoChunks(
  text: string,
  opts?: { chunkSize?: number; overlap?: number }
): string[] {
  const chunkSize = opts?.chunkSize ?? 400;
  const overlap = opts?.overlap ?? 80;

  const words = text.split(/\s+/);
  if (words.length <= chunkSize) return [text];

  const chunks: string[] = [];
  let start = 0;

  while (start < words.length) {
    const end = Math.min(start + chunkSize, words.length);
    chunks.push(words.slice(start, end).join(" "));
    if (end >= words.length) break;
    start += chunkSize - overlap;
  }

  return chunks;
}

File: python/agent_vfs/search/chunker.py
from __future__ import annotations


def split_into_chunks(
    text: str,
    *,
    chunk_size: int = 400,
    overlap: int = 80,
) -> list[str]:
    """Split text into overlapping chunks for embedding.

    Uses whitespace-based token approximation.
    """
    words = text.split()
    if len(words) <= chunk_size:
        return [text]

    chunks: list[str] = []
    start = 0

    while start < len(words):
        end = min(start + chunk_size, len(words))
        chunks.append(" ".join(words[start:end]))
        if end >= len(words):
            break
        start += chunk_size - overlap

    return chunks
